### PR TITLE
Updating existing tests with updated syntax and validation checking.

### PR DESCRIPTION
**Pull request by StephenBonikowsky**
* Using Asserts instead of StringBuilder for validation.
* Adding comments to compartmentalize the test into separate sections.

### CODE REVIEW

**Comment by StephenBonikowsky** [comment]
No functionality should have changed.

Anybody know if when an Assert fails will Xunit include the inner exception details when it throws?

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs [hunk 1]
```diff
@@ -19,25 +19,35 @@ public static class ServiceContractTests
     public static void DefaultSettings_Echo_RoundTrips_String_Buffered()
```
Comment (iamjasonp) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs]: Maybe these could be consolidated into a [Theory] test? It looks like much of this is repeated code.

**Comment by iamjasonp** [comment]
The inner exception shows when a throw happens in a `[Fact]` test. 
Not sure about `[Theory]`, or within an `Assert.Throws<T>` block

File: src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs (hunk 1 above)
Comment (StephenBonikowsky) [inline src/System.Private.ServiceModel/tests/Scenarios/Contract/Service/ServiceContractTests.cs]: Possibly, don't really want to get bogged down refactoring these tests, just some quick cleanup.

**Pull request merged**